plyara/utils.py

```diff
@@ -87,14 +87,10 @@ def detect_imports(rule):
     for imp in Parser.IMPORT_OPTIONS:
         imp_module = '{}.'.format(imp)
 
-        if imp in condition_terms and imp not in detected_imports:
-            detected_imports.append(imp)
-
-        elif imp not in detected_imports:
-            for term in condition_terms:
-                if term.startswith(imp_module):
-                    detected_imports.append(imp)
-                    break
+        for term in condition_terms:
+            if term.startswith(imp_module):
+                detected_imports.append(imp)
+                break
```

Comment:
The logic in the check imports had some code that didn't make sense. The function is meant to detect what imports are needed based on the conditions in the rule, so it's rewritten to just do that. It was adding imports directly to the detected imports without checking if they're in the conditions.

Reply:
interesting. we should be sure to document this behavior change somewhere, just in case.

Reply:
The part I removed I don't think will ever be executed. The variable that is looked for in `condition_terms` is `imp` which is the bare import. That's never going to be found in the `condition_terms` unless I'm wrong. Also, the `elif` after that has no effect. The loop is based on the list of imports which has no duplicates. This check is not needed since each import is searched once and if found it will break.

Reply:
Basically, I don't see a reason to change documentation since I don't see any condition that will be different after this code simplification.

Reply:
Ah, I misunderstood then, I thought this was changing to only include *used* imports rather than all listed imports.